onnxruntime/core/optimizer/selectors_actions/helpers.h

```diff
@@ -5,20 +5,49 @@
 
 namespace onnxruntime {
 
+// if the last input in num_inputs is for the variadic input, the variadic input could have zero or more values
+// so we need to special case the zero and count that as one. same for outputs
+inline int NumIOEntries(bool variadic_io, int num_io, int num_variadic_io) {
+  return variadic_io
+             ? num_io + std::max(1, num_variadic_io) - 1
+             : num_io;
+}
+
 //
 // Selection helpers
 //
 
-// Struct to serialize the node indexes in an ORT format model.
+// Struct to serialize the node indices in an ORT format model.
 // Use EmptyNodeIndex for nullptr entries in the vectors for missing optional inputs
-struct NodesToOptimizeIndexes {
-  std::vector<NodeIndex> nodes;
-  int num_inputs;
-  int num_outputs;
-  bool variadic_input;
-  bool variadic_output;
-  int num_variadic_inputs;
-  int num_variadic_outputs;
+struct NodesToOptimizeIndices {
+  const std::vector<NodeIndex> nodes_;
```

Comment:
>nodes_ [](http://example.com/codeflow?start=31&length=6)

struct data member names generally don't end with "_"
[https://google.github.io/styleguide/cppguide.html#Variable_Names](https://google.github.io/styleguide/cppguide.html#Variable_Names)

that would make the existing ctor more annoying to write.

but maybe we can skip it and use aggregate initialization if we deal with NodesToOptimizeIndices directly instead of through std::unique_ptr.

could we have an interface like this instead?
`bool NodeSelector::Select(const GraphViewer& graph_viewer, const Node& node, std::unique_ptr<NodesToOptimizeIndices>& selection) const` ->
`std::optional<NodesToOptimizeIndices> NodeSelector::Select(const GraphViewer& graph_viewer, const Node& node) const`